### PR TITLE
Adding environment-name option

### PR DESCRIPTION
**Pull request by tdegeus**

### CODE REVIEW

**Review by beckermr — CHANGES_REQUESTED** [review]
Don't change the dist files. You need to adjust the source ones IIRC.

**Comment by tdegeus** [comment]
> Don't change the dist files. You need to adjust the source ones IIRC.

I guess I was too quick ;)

File: .github/workflows/options.yml [hunk 1]
```diff
@@ -0,0 +1,38 @@
+name: options
+on:
+  push:
+    branches:
+      - main
+  pull_request: null
```
Comment (tdegeus) [inline .github/workflows/options.yml:6]: Should we change this to run CI on the new option?

File: index.js [hunk 2]
```diff
@@ -43,7 +43,7 @@ async function run () {
     const envFileName = core.getInput('environment-file')
     const envFilePath = path.join(process.env.GITHUB_WORKSPACE || '', envFileName)
     const envYaml = yaml.safeLoad(fs.readFileSync(envFilePath, 'utf8'))
-    const envName = envYaml.name
+    const envName = core.getInput('environment-name') || envYaml.name
```
Comment (beckermr) [inline index.js:46]: This var is only used for outputting. You'll have to pass it to micromamba to get the right env name there too.

File: .github/workflows/options.yml (hunk 1 above)
Comment (beckermr) [reply .github/workflows/options.yml:6]: yes we want to run CI with the new options

Comment (beckermr) [reply .github/workflows/options.yml:6]: I suspect it is not running now because the PR is from a fork.

File: index.js (hunk 2 above)
Comment (tdegeus) [reply index.js:46]: My bad, it should be ok with the latest commit

File: .github/workflows/options.yml (hunk 1 above)
Comment (tdegeus) [reply .github/workflows/options.yml:6]: It is in fact not entirely clear to me how to resolve this

Comment (tdegeus) [reply .github/workflows/options.yml:6]: I think you have to enable something : https://github.blog/2020-08-03-github-actions-improvements-for-fork-and-pull-request-workflows/

Comment (tdegeus) [reply .github/workflows/options.yml:6]: Or maybe even a repo owner @wolfv

**Comment by tdegeus** [comment]
Sorry I missed the separate powershell implementation

**Comment by tdegeus** [comment]
In fact I should just fix my mistakes rather than blaming GH for not running the action

File: dist/index.js [hunk 3]
```diff
@@ -110,7 +110,7 @@ async function run () {
 
       // final bits of the install
       await execute('mkdir -p ' + path.join(os.homedir(), 'micromamba/pkgs/'))
-      await execute('source ' + profile + ' && micromamba create --strict-channel-priority -y -f ' + envFilePath)
+      await execute('source ' + profile + ' && micromamba create -n ' + envName + ' --strict-channel-priority -y -f ' + envFilePath)
       fs.appendFileSync(profile, 'set -eo pipefail\n')
```
Comment (wolfv) [inline dist/index.js:114]: hmm, is this manually edited? I think you need to edit the `index.js` that's in the root folder, and then build with nodejs (can get it from conda-forge) usign 

```
mamba install nodejs -c conda-forge
npm install
npm run build
```

Comment (tdegeus) [reply dist/index.js:114]: Ah yeah, I had both files open and didn't look well. What should I do with changes to `package-lock.json`?

**Comment by tdegeus** [comment]
Good to go I think!

File: dist/index.js (hunk 3 above)
Comment (beckermr) [reply dist/index.js:114]: Commit those changes.

**Review by beckermr — CHANGES_REQUESTED** [review]
For the tests of the options, they should check that the activate conda env is named correctly.

**Comment by tdegeus** [comment]
@wolfv How can I check if the environment loaded is the one I specified. I wanted to grep from `micromamba info` but it seems that it does not exist

**Comment by beckermr** [comment]
May check an env var like `CONDA_PREFIX` or something?

**Review by beckermr — APPROVED** [review]

**Comment by beckermr** [comment]
Thanks a bunch, @tdegeus!

**Comment by tdegeus** [comment]
Awesome! Can't wait to start using it!

**Comment by beckermr** [comment]
Yup. You'll need to use version 8 once we tag it.

**Pull request merged**